### PR TITLE
`Animator` compoment add default state ability

### PR DESCRIPTION
**Pull request by luzhuang**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### Other information:

### CODE REVIEW

File: packages/core/src/animation/Animator.ts [hunk 1]
```diff
@@ -776,6 +779,20 @@ export class Animator extends Component {
...
 
+  private _onAnimatorControllerChange() {
+    this._checkAutoPlay();
+  }
+
+  private _checkAutoPlay() {
+    const { layers } = this._animatorController;
+    for (let i = 0, n = layers.length; i < n; ++i) {
+      const stateMachine = layers[i].stateMachine;
+      if (stateMachine && stateMachine.entryState) {
```
Comment (GuoLei1990) [inline packages/core/src/animation/Animator.ts:792]: stateMachine?.entryState

File: packages/core/src/animation/AnimatorStateMachine.ts [hunk 2]
```diff
@@ -10,6 +10,9 @@ export class AnimatorStateMachine {
   /** The list of states. */
   readonly states: AnimatorState[] = [];
 
+  /** The state will be played automatically */
+  entryState: AnimatorState;
```
Comment (GuoLei1990) [inline packages/core/src/animation/AnimatorStateMachine.ts:14]: I like `defaultState`, means the default play state.

comment must add `.` and not clearly

File: packages/core/src/animation/AnimatorController.ts [hunk 3]
```diff
@@ -64,7 +67,8 @@ export class AnimatorController {
     return this._updateFlagManager.createFlag(BoolUpdateFlag);
   }
 
-  private _distributeUpdateFlag(): void {
+  private _onChange() {
```
Comment (GuoLei1990) [inline packages/core/src/animation/AnimatorController.ts:70]: add `void`

File: packages/core/src/animation/Animator.ts [hunk 4]
```diff
@@ -59,10 +59,13 @@ export class Animator extends Component {
...
-    if (animatorController !== this._animatorController) {
+    if (animatorController && animatorController !== this._animatorController) {
       this._controllerUpdateFlag && this._controllerUpdateFlag.destroy();
       this._controllerUpdateFlag = animatorController && animatorController._registerChangeFlag();
+      this._animatorController && (this._animatorController._onDataChanged = null);
       this._animatorController = animatorController;
+      this._onAnimatorControllerChange();
+      animatorController._onDataChanged = this._onAnimatorControllerChange.bind(this);
     }
   }
```
Comment (GuoLei1990) [inline packages/core/src/animation/Animator.ts:71]: animatorController is shared, this have bug

File: packages/core/src/animation/AnimatorStateMachine.ts (hunk 2 above)
Comment (luzhuang) [reply packages/core/src/animation/AnimatorStateMachine.ts:14]: The editor named entryState, i think it should be samed

File: packages/loader/src/index.ts [hunk 5]
```diff
@@ -13,11 +13,11 @@ import "./SpriteAtlasLoader";
...
 import "./TextureCubeLoader";
+import "./AnimatorControllerLoader";
 
 export { parseSingleKTX } from "./compressed-texture";
 export { GLTFResource } from "./gltf/GLTFResource";
 export type { GLTFParams } from "./GLTFLoader";
 export * from "./resource-deserialize";
 export * from "./SceneLoader";
 export type { Texture2DParams } from "./Texture2DLoader";
-
```
Comment (GuoLei1990) [inline packages/loader/src/index.ts:23]: revert this file

File: packages/core/src/animation/Animator.ts [hunk 6]
```diff
@@ -776,6 +780,16 @@ export class Animator extends Component {
...
 
+  private _checkAutoPlay() {
```
Comment (GuoLei1990) [inline packages/core/src/animation/Animator.ts:783]: add `:void`

File: packages/core/src/animation/Animator.ts [hunk 7]
```diff
@@ -59,10 +59,11 @@ export class Animator extends Component {
   }
 
   set animatorController(animatorController: AnimatorController) {
-    if (animatorController !== this._animatorController) {
+    if (animatorController && animatorController !== this._animatorController) {
       this._controllerUpdateFlag && this._controllerUpdateFlag.destroy();
       this._controllerUpdateFlag = animatorController && animatorController._registerChangeFlag();
       this._animatorController = animatorController;
+      this._reset();
```
Comment (GuoLei1990) [inline packages/core/src/animation/Animator.ts:66]: I check `_reset` func, maybe use `this._animationCurveOwners`  to revert is better

File: packages/core/src/animation/Animator.ts [hunk 8]
```diff
@@ -59,10 +59,11 @@ export class Animator extends Component {
   }
 
   set animatorController(animatorController: AnimatorController) {
-    if (animatorController !== this._animatorController) {
+    if (animatorController && animatorController !== this._animatorController) {
```
Comment (GuoLei1990) [inline packages/core/src/animation/Animator.ts:62]: `animatorController &&` this logic maybe no useful?

File: packages/core/src/animation/AnimatorStateMachine.ts (hunk 2 above)
Comment (GuoLei1990) [reply packages/core/src/animation/AnimatorStateMachine.ts:14]: OK! Let me see see editor!

File: packages/loader/src/index.ts (hunk 5 above)
Comment (luzhuang) [reply packages/loader/src/index.ts:23]: it is formatted by vscode ..

**Pull request merged**